### PR TITLE
e2e/framework/metrics: handle index out of bounds panic

### PR DESCRIPTION
**Pull request by aroradaman**
#### What type of PR is this?
This handles a potential panic in e2e kube proxy metrics collector.
/kind bug

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

/sig network

### CODE REVIEW

File: test/e2e/framework/metrics/kube_proxy_metrics.go [hunk 1]
```diff
@@ -25,7 +25,10 @@ type KubeProxyMetrics testutil.Metrics
 
 // GetCounterMetricValue returns value for metric type counter.
 func (m *KubeProxyMetrics) GetCounterMetricValue(metricName string) float64 {
-	return float64(testutil.Metrics(*m)[metricName][0].Value)
+	if len(testutil.Metrics(*m)[metricName]) > 0 {
+		return float64(testutil.Metrics(*m)[metricName][0].Value)
+	}
+	return 0
```
[the remark this answers is not in the document]
Comment (aojea) [reply test/e2e/framework/metrics/kube_proxy_metrics.go:31]: Or return an error in addition to the value and make the caller fail

File: test/e2e/framework/metrics/kube_proxy_metrics.go [hunk 2]
```diff
@@ -17,15 +17,19 @@ limitations under the License.
...
 )
 
 // KubeProxyMetrics is metrics for kube-proxy
 type KubeProxyMetrics testutil.Metrics
 
 // GetCounterMetricValue returns value for metric type counter.
-func (m *KubeProxyMetrics) GetCounterMetricValue(metricName string) float64 {
-	return float64(testutil.Metrics(*m)[metricName][0].Value)
+func (m *KubeProxyMetrics) GetCounterMetricValue(metricName string) (float64, error) {
+	if len(testutil.Metrics(*m)[metricName]) > 0 {
```
Comment (richabanker) [inline test/e2e/framework/metrics/kube_proxy_metrics.go:29]: Nit: This reads a bit opposite to the general way of error checking in Go. Usually we return if err != nil ( in this case of len(metrics) == 0) and then resume with the happy path.

**Comment by aojea** [comment]
/lgtm
/approve 

Thanks

The ipvs job seems to be passing on the periodic https://testgrid.k8s.io/sig-network-gce#gci-gce-ipvs and the errors are about pod timeouts

**Pull request merged**